BE/src/interfaces/challenge.ts

```diff
@@ -1,18 +1,16 @@
-// export enum StatusState {
-//   Progress = 0,
-//   Success,
-//   Failure
-// }
+export enum StatusState {
+  Progress = 0,
+  Success,
+  Failure
+}
 
 export interface IChallenge {
   id: number;
-  state: string;
+  state: StatusState;
   startDate: string;
-  tasksOrder: string;
+  tasksOrder: Set<string>;
   tasksStatus: string;
-  achievementsStatus: string;
+  achievementsStatus: Map<number, string>;
 }
 
-export type startNewChallengeType = {
-  (id: number): IChallenge,
-};
+export type startNewChallengeType = (id: number) => IChallenge;
```

Comment:
As I wrote earlier, it has more than 1 parameter:
![image](https://user-images.githubusercontent.com/68462576/121855609-394b6780-ccfc-11eb-9924-c7bdda12a9dc.png)